runtime/bcutil/cfreader.c

```diff
@@ -3601,3 +3606,18 @@ sortMethodIndex(J9CfrConstantPoolInfo* constantPool, J9CfrMethod *list, IDATA st
 		}
 	}
 }
+
+#if JAVA_SPEC_VERSION >= 15
+I_32
+checkClassBytes(J9VMThread *currentThread, U_8* classBytes, UDATA classBytesLength, U_8* segment, U_32 segmentLength) 
+{
+	I_32 rc = 0;
+	U_32 cfrFlags = BCT_JavaMaxMajorVersionShifted | BCT_AnyPreviewVersion | BCT_BasicCheckOnly;
+	PORT_ACCESS_FROM_VMC(currentThread);
+	if (NULL != classBytes) {
+		rc = j9bcutil_readClassFileBytes(PORTLIB, NULL, classBytes, classBytesLength, segment, segmentLength, cfrFlags, NULL, NULL, 0, 0);
```

Comment:
The first `NULL` argument could be changed to pass a `J9CfrClassFile` struct to fill in which makes getting the `thisClass` field easy:

```
J9CfrClassFile *classfile = .....;
	U_32 cpThisClassUTF8Slot = constantPool[classfile->thisClass].slot1;
	U_32 originalStringLength = constantPool[cpThisClassUTF8Slot].slot1;
	const char* originalStringBytes = (const char*)constantPool[cpThisClassUTF8Slot].bytes;
....
```

Reply:
I see the second parameter of j9bcutil_readClassFileBytes() takes a function pointer, can I pass in a J9CfrClassFile struct ?

Reply:
Sorry, I misread the function descriptor.  The J9CfrClassFile struct is written into the buffer - see `ClassFileParser::parseClassFile` for an example:

```c
if ( BCT_ERR_NO_ERROR == result ) {
		/* return the buffer */
		_j9CfrClassFile = (J9CfrClassFile *)buffer;
```